charts/ibm-fhir-server/templates/deployment.yaml

```
@@ -65,44 +95,31 @@ spec:
             - name: WLP_LOGGING_JSON_FIELD_MAPPINGS
               value: loglevel:level
             - name: FHIR_DB_HOSTNAME
-              value: "{{ .Values.db.host }}"
+              value: "{{ include "fhir.database.host" $ }}"
             - name: FHIR_DB_PORT
-              value: "{{ .Values.db.port }}"
+              value: "{{ include "fhir.database.port" $ }}"
             - name: FHIR_DB_NAME
-              value: "{{ .Values.db.name }}"
+              value: "{{ include "fhir.database.name" $ }}"
             - name: FHIR_DB_USER
-              value: "{{ .Values.db.user }}"
-            {{- if .Values.db.dbSecret }}
-            {{- if .Values.db.passwordSecretKey }}
+              value: "{{ include "fhir.database.user" $ }}"
             - name: FHIR_DB_PASSWORD
               valueFrom:
                 secretKeyRef:
-                  name: {{ .Values.db.dbSecret }}
-                  key: {{ .Values.db.passwordSecretKey }}
-            {{- end }}
+                  name: "{{ include "fhir.database.secretName" $ }}"
+                  key: "{{ include "fhir.database.secretKey" $ }}"
```

```suggestion
                  key: "{{ include "fhir.database.passwordSecretKey" $ }}"
```

per suggestion in _helpers.tpl